### ISSUE TITLE
New `filter-length` name isn't descriptive enough

### ISSUE THREAD

**Issue by MatthewDaggitt** [agda/agda-stdlib#234]
Need room for things like
```agda
∀ xs → length (filter (P?⇒¬P? P?) xs) + length (filter P? xs) ≡ length xs
∀ xs → Any (λ x → ¬ (P x)) xs → length (filter P? xs) < length xs
∀ {xs} → length (filter P? xs) ≡ length xs →filter P? xs ≡ xs
```

**Comment by gallais** [comment]
Not fully convinced myself but we could use `length-filter-≤` and for the others have
things like:

```agda
length-partition : ∀ xs → length (filter (P?⇒¬P? P?) xs) + length (filter P? xs) ≡ length xs
length-filter-< : ∀ xs → Any (λ x → ¬ (P x)) xs → length (filter P? xs) < length xs
length-filter-≡-inv : ∀ {xs} → length (filter P? xs) ≡ length xs →filter P? xs ≡ xs
```
The last one is an inversion lemma rather than say:

```agda
length-filter-≡ : ∀ xs → All P xs → length (filter P? xs)  ≡ length xs
```

**Comment by MatthewDaggitt** [comment]
You're completely right. Just had another look and I realised that part of the problem was that the names of the lemmas previously added were actually the wrong way round. Now that those names have been fixed, the names for the new lemmas are obvious (see d2ecbdf).

I've left the `length-filter-≡` lemma out for now, as hopefully that'll be an immediate consequence if I get round to pushing my permutation stuff in a future release.
